lara-typescript/src/section-authoring/components/text-block-edit-form.tsx

```diff
@@ -1,34 +1,60 @@
 import * as React from "react";
-import { useRef } from "react";
+import { useCallback, useRef } from "react";
 import { Editor } from "@tinymce/tinymce-react";
 import { ISectionItem, ITextBlockData } from "../api/api-types";
 import { usePageAPI } from "../hooks/use-api-provider";
+import { debounce } from "ts-debounce";
 
 import "./text-block-edit-form.scss";
 
 export interface ITextBlockEditFormProps {
   pageItem: ISectionItem;
+  handleUpdateItemPreview?: (updates: Record<string, any>) => void;
 }
 
 export const TextBlockEditForm: React.FC<ITextBlockEditFormProps> = ({
-  pageItem
+  pageItem, handleUpdateItemPreview
   }: ITextBlockEditFormProps) => {
   const { content, name, isCallout, isHalfWidth } = pageItem?.data as ITextBlockData;
   const editorRef = useRef<any>(null);
   const initEditor = (e: any, editor: any) => {
     editorRef.current = editor;
   };
   const contentRef = useRef<HTMLTextAreaElement|null>(null);
+  const updates = useRef<Partial<ISectionItem>>({});
   const api = usePageAPI();
   const pathToTinyMCE = api.pathToTinyMCE;
   const pathToTinyMCECSS = api.pathToTinyMCECSS;
 
+  const _updatePreview = (update: any) => {
+    if (handleUpdateItemPreview) {
+      updates.current = {...updates.current, ...update};
+      handleUpdateItemPreview(updates.current);
+    }
+  };
+
+  const updatePreview = useCallback(debounce(_updatePreview, 500), []);
```

Comment:
Hmn, I haven't used `useCallback` much myself, so I went and read the documentation, but I guess I still don't understand this.

https://reactjs.org/docs/hooks-reference.html#usecallback